pkg/api/types.go

```diff
@@ -673,6 +673,9 @@ type GlusterfsVolumeSource struct {
 	// Required: EndpointsName is the endpoint name that details Glusterfs topology
 	EndpointsName string `json:"endpoints"`
 
+	// Optional: Servers is the gluster server list
+	EndpointsNameSpace *string `json:"endpointsns,omitempty"`
```

Comment:
Is converting the endpoint name/namespace into a DNS name and resolving IP addresses all that the platform does? Is it all it will ever do? Could this allow end runs around network plugins that try to limit a pod to only contacting services and pods within its namespace?